### PR TITLE
update API docs to specify how to set content encoding

### LINKED ISSUE #32: Update API docs to specify how to set content encoding
The `submitBlock` API supports optional GZIP compression for the request body.

Right now, the API docs only indicate how to hint at content type (`Content-Type: {application/json, application/octet-stream}`).

Following the relay implementation from `flashbots/mev-boost-relay`, it seems the preference is to support the optional header: `Content-Encoding` with a single allowed value `gzip`.

The API docs should reflect this so implementers know how to support this feature.

### PR DESCRIPTION
**Pull request by intldds**
closes #32

let me know if you need me to change/add anything :)

### CODE REVIEW

File: apis/relay/builder/blocks.yaml [hunk 1]
```diff
@@ -7,6 +7,9 @@ post:
       content type should be `application/json`. If SSZ, the content type should
       be `application/octet-stream`.
     
+    * To enable GZIP compression for the request body, the preference is to 
+    support the optional header `Content-Encoding` with a single allowed value `gzip`.
```
Comment (ralexstokes) [inline apis/relay/builder/blocks.yaml:11]: I would say something instead like
```suggestion
    * To enable GZIP compression for the request body, the HTTP content encoding should be `gzip`. Compression is optional.
```

File: apis/relay/builder/blocks.yaml [hunk 2]
```diff
@@ -47,6 +50,8 @@ post:
       application/octet-stream:
         schema:
           description: "SSZ serialized request. Use content type header to indicate that SSZ data is contained in the request body."
+        encoding:
+          Content-Encoding: gzip
```
Comment (ralexstokes) [inline apis/relay/builder/blocks.yaml:54]: following the examples here: https://spec.openapis.org/oas/latest.html#considerations-for-file-uploads

it looks like we want this instead
```suggestion
          contentEncoding: gzip
```

and also declared under content.application/json.schema

**Review by ralexstokes — CHANGES_REQUESTED** [review]
great thanks! need some updates but then I think this is good to merge :)

File: apis/relay/builder/blocks.yaml (hunk 2 above)
Comment (intldds) [reply apis/relay/builder/blocks.yaml]: thanks for helping me! new commit made

**Comment by jtraglia** [comment]
Hey @intldds, thanks for the PR! This has an error if you check the linter; see the CI errors. You can fix that problem by putting `contentEncoding` in a new `encoding` section, like:

```yaml
      application/octet-stream:
        schema:
          description: "SSZ serialized request. Use content type header to indicate that SSZ data is contained in the request body."
        encoding:
          contentEncoding: gzip
```

But then there's another error which I don't really understand. Not sure how to fix:

```
 51:28  error  oas3-schema  "contentEncoding" property type must be object.  post.requestBody.content.application/octet-stream.encoding.contentEncoding
```

**Comment by intldds** [comment]
Thanks for the reply @jtraglia! I've tried following the examples provided here (https://spec.openapis.org/oas/latest.html#considerations-for-file-uploads) and I've tried different combinations to place `contentEncoding`.

In all of them, when I ran `spectral lint relay-oapi.yaml` I always received "Property "contentEncoding" is not expected to be here" -- feel free to reject the PR in this case.

**Comment by jtraglia** [comment]
I can't figure it out either 😭 let's just remove the `encoding` sections and keep the bullet point.

**Review by ralexstokes — APPROVED** [review]
this will do, thanks @intldds !

**Review by jtraglia — APPROVED** [review]
LGTM 👍 Thanks!

**Pull request merged**